athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/propertygraph/rowwriters/EdgeRowWriter.java

```diff
@@ -167,4 +190,32 @@ else if (fieldName.equals(SpecialKeys.OUT.toString().toLowerCase())) {
                 break;
         }
     }
+
+    private static Map<String, Object> contextAsMap(Object context, boolean caseInsensitive)
+    {
+        Map<String, Object> contextAsMap = (Map<String, Object>) context;
+
+        if (!caseInsensitive) {
+            return contextAsMap;
+        }
+
+        Object fieldValueID = contextAsMap.get(T.id);
+        if (fieldValueID != null) {
+            Object fieldValueIN = ((LinkedHashMap) contextAsMap.get(Direction.IN)).get(T.id);
+            Object fieldValueOUT = ((LinkedHashMap) contextAsMap.get(Direction.OUT)).get(T.id);
+
+            contextAsMap.remove(T.id);
+            contextAsMap.remove(T.label);
+            contextAsMap.remove(Direction.IN);
+            contextAsMap.remove(Direction.OUT);
+
+            contextAsMap.put(SpecialKeys.ID.toString(), fieldValueID);
+            contextAsMap.put(SpecialKeys.IN.toString(), fieldValueIN);
+            contextAsMap.put(SpecialKeys.OUT.toString(), fieldValueOUT);         
```

Comment:
Just checking, is it safe to mutate `Object context` here?

Meaning, is this object ever used beyond the scope of just the lambda where `contextAsMap` is called?

If so, you may want to perform these mutations on the TreeMap copy of this object instead.

I'm not sure though, that's why I'm just asking to double check.

Reply:
The tree map copy fails because, the keys are not String in the original context map. The first call to the context map per record, transforms the records else it will keep failing. And then the next call just use the modified context map. But this is limited to per record processing

Reply:
Sounds fine to me. Just wanted to double check.